apps/greencheck/templates/backfill_by_url_procedure.sql

```diff
@@ -0,0 +1,51 @@
+
+-- then for each url, we run a query to get the latest result from
+-- greencheck using the date index, and if it is green
+-- load that into the green_presenting table
+
+DROP PROCEDURE IF EXISTS backfill_by_url;
+
+CREATE PROCEDURE backfill_by_url()
+
+BEGIN
+  -- we need a way to check so we can exit the loop
+  DECLARE done INT(8) DEFAULT FALSE;
+  DECLARE counter INT(8) DEFAULT FALSE;
+
+  -- declare the types we'd fetch from the cursor
+  DECLARE fetched_url                VARCHAR(255);
+
+  -- set up our cursor, so we have a list of urls to loop through
+  DECLARE cur CURSOR FOR
+  SELECT url FROM urls;
```

Comment:
From what I understand the urls table is a temporary table that you've filled from a csv. From where does that CSV come from? Is it a dump of all urls?

Reply:
It's a dump of all the urls, once I've filtered out the ones that weren't valid hostnames.

I did indeed create a temporary table for this.

My thinking is that it lets us either parallelize the work, or be able to pick up where we left off.

Reply:
If it is a dump of all the urls, are they unique? If they are we'll reduce the amount of urls. Perhaps we could reduce the space even further by only including one url with the domain. For instance

- sub.example.com
- sub2.example.com
- www.example.com
- www.example.com/hello

This would only show up once in that table.

Reply:
Alternatively we could populate the url table with this -> https://gist.github.com/chilts/7229605 and then just rely on the stored procedure that automatically inputs new urls in the green_presenting table. 

After all "Perfect is the enemy of good", and that will get us a green2url zip that we can slowly expand upon over time.